### PR TITLE
updated: validations and api-doc

### PR DESCRIPTION
**Pull request by nevil-mathew**

### CODE REVIEW

File: src/services/default-rule.js [hunk 1]
```diff
@@ -42,61 +52,58 @@ module.exports = class DefaultRuleHelper {
...
-			})
+			errors.push({ param: 'requester_field', msg: 'Invalid requester_field' })
 		}
 
-		if (errors.length !== 0) {
-			return {
-				isValid: false,
-				errors,
+		if (validTargetField.length > 0 && validRequesterField.length > 0) {
+			if (validTargetField[0]?.data_type !== validRequesterField[0]?.data_type) {
```
Comment (rakeshSgr) [inline src/services/default-rule.js:73]: @nevil-mathew if validTargetField has  more than one field, then won't check for those ?

File: src/services/default-rule.js [hunk 2]
```diff
@@ -42,61 +52,58 @@ module.exports = class DefaultRuleHelper {
...
+				errors.push({
+					param: 'target_field,requester_field',
+					msg: 'Data types of target_field and requester_field should match',
+				})
+			} else {
+				const operatorValidation = {
```
Comment (rakeshSgr) [inline src/services/default-rule.js:79]: @nevil-mathew checking for first field is enough?

Comment (nevil-mathew) [reply src/services/default-rule.js:79]: We're checking if the target_field and requester_field have the same data_type before this, when we reach this condition both will have same data_type

File: src/services/default-rule.js (hunk 1 above)
Comment (nevil-mathew) [reply src/services/default-rule.js:73]: We're only checking other conditions if the target and requester fields are valid.
Else we don't need to check the data type, operator validity.

**Pull request merged**